src/ObjCRuntime/Messaging.cs

```diff
@@ -1,5 +1,190 @@
+		// An additional IntPtr.Zero is always passed at the end to terminate any varargs arrays.
+		internal static IntPtr objc_msgSend_6_vargs (IntPtr arg1, IntPtr arg2, IntPtr arg3, IntPtr arg4, IntPtr arg5, IntPtr arg6, params IntPtr [] varArguments)
+		{
+			return CallObjCMsgSendVarArgs (ShuffleVarArgs (6, arg1, arg2, arg3, arg4, arg5, arg6, IntPtr.Zero, IntPtr.Zero, varArguments));
+		}
+
+		// Call objc_msgSend with 7 non-variadic arguments, and up to 10 (MaxVarArgs) variadic arguments.
+		// An additional IntPtr.Zero is always passed at the end to terminate any varargs arrays.
+		internal static IntPtr objc_msgSend_7_vargs (IntPtr arg1, IntPtr arg2, IntPtr arg3, IntPtr arg4, IntPtr arg5, IntPtr arg6, IntPtr arg7, params IntPtr [] varArguments)
+		{
+			return CallObjCMsgSendVarArgs (ShuffleVarArgs (7, arg1, arg2, arg3, arg4, arg5, arg6, arg7, IntPtr.Zero, varArguments));
+		}
+
+		// Call objc_msgSend with 8 non-variadic arguments, and up to 10 (MaxVarArgs) variadic arguments.
+		// An additional IntPtr.Zero is always passed at the end to terminate any varargs arrays.
+		internal static IntPtr objc_msgSend_8_vargs (IntPtr arg1, IntPtr arg2, IntPtr arg3, IntPtr arg4, IntPtr arg5, IntPtr arg6, IntPtr arg7, IntPtr arg8, params IntPtr [] varArguments)
+		{
+			return CallObjCMsgSendVarArgs (ShuffleVarArgs (8, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, varArguments));
+		}
+
+		static internal int MaxVarArgs = 10;
+
+		static IntPtr CallObjCMsgSendVarArgs (IntPtr [] shuffledArguments)
+		{
+			return CallVariadicFunction (objc_msgSend_variadic, shuffledArguments);
+		}
+
+		[DllImport (LIBOBJC_DYLIB, EntryPoint = "objc_msgSend")]
+		extern static IntPtr objc_msgSend_variadic (IntPtr arg1, IntPtr arg2, IntPtr arg3, IntPtr arg4, IntPtr arg5, IntPtr arg6, IntPtr arg7, IntPtr arg8, IntPtr arg9, IntPtr arg10, IntPtr arg11, IntPtr arg12, IntPtr arg13, IntPtr arg14, IntPtr arg15, IntPtr arg16, IntPtr arg17, IntPtr arg18, IntPtr arg19);
+
+		// This method creates an array of normal arguments to pass to objc_msgSend given the specified normal arguments + varargs arguments.
+		static IntPtr [] ShuffleVarArgs (int argCount, IntPtr arg1, IntPtr arg2, IntPtr arg3, IntPtr arg4, IntPtr arg5, IntPtr arg6, IntPtr arg7, IntPtr arg8, params IntPtr [] varArguments)
+		{
+			if (varArguments.Length > MaxVarArgs)
+				throw new ArgumentOutOfRangeException ($"A maximum of {MaxVarArgs} variadic arguments is supported.");
+
+			var rv = new IntPtr [8 + MaxVarArgs];
+
+			rv [0] = arg1;
+			rv [1] = arg2;
+			rv [2] = arg3;
+			rv [3] = arg4;
+			rv [4] = arg5;
+			rv [5] = arg6;
+			rv [6] = arg7;
+			rv [7] = arg8;
+
+			// ARM64: 8 arguments in registers, the rest is on the stack. This is where iOS/ARM64 expects the first varargs arguments.
+			var varArgsStartIndex = Runtime.IsARM64CallingConvention ? 8 : argCount;
+
+			for (var i = 0; i < varArguments.Length; i++) {
+				rv [varArgsStartIndex + i] = varArguments [i];
+			}
```

Comment:
Why not array.copy here using the start index: https://learn.microsoft.com/en-us/dotnet/api/system.array.copy?view=net-8.0#system-array-copy(system-array-system-int64-system-array-system-int64-system-int64)

Reply:
Changed.